src/main/java/ro/msg/learning/shop/service/ProductService.java

```diff
@@ -0,0 +1,102 @@
+package ro.msg.learning.shop.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.dto.ProductDto;
+import ro.msg.learning.shop.entity.Product;
+import ro.msg.learning.shop.entity.ProductCategory;
+import ro.msg.learning.shop.entity.Supplier;
+import ro.msg.learning.shop.repository.ProductCategoryRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.SupplierRepository;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class ProductService {
+
+    private final ProductRepository productRepository;
+    private final ProductCategoryRepository productCategoryRepository;
+    private final SupplierRepository supplierRepository;
+
+    public List<Product> findAll() {
```

Comment:
Your service methods should return Dtos (and also should receive Dtos as parameters). In application layer (Controllers) it is not recommended to have access to the Data Model classes, that's why we have Dtos (to expose data in UI through controllers).